reactor-core/src/main/java/reactor/core/publisher/OnNextFailureStrategy.java

```diff
@@ -0,0 +1,274 @@
+	 * @param valueConsumer the {@link Consumer} to process the error-causing values with.
+	 * It must deal with potential {@code null}s.
+	 * @return a new {@link OnNextFailureStrategy} that allows resuming the sequence.
+	 */
+	static OnNextFailureStrategy resume(Consumer<Throwable> errorConsumer,
+			Consumer<Object> valueConsumer) {
+		return new ResumeStrategy(null, errorConsumer, valueConsumer);
+	}
+
+	/**
+	 * A strategy that let some errors resume. When processing, the error and the
+	 * incriminating source value are passed to custom {@link Consumer Consumers}.
+	 * <p>
+	 * Any exception thrown by the predicate is thrown as is. Any exception thrown by
+	 * the consumers will suppress the original error and be returned for propagation.
+	 * Even if the original error is fatal, if it passes the predicate then it can
+	 * be processed and recovered from (see {@link Exceptions#throwIfFatal(Throwable)}).
+	 *
+	 * @param causePredicate the {@link Predicate} to use to determine if a failure
+	 * should be recovered from.
+	 * @param errorConsumer the {@link Consumer} to process the recovered errors with.
+	 * @param valueConsumer the {@link Consumer} to process the error-causing values with.
+	 * It must deal with potential {@code null}s.
+	 * @return a new {@link OnNextFailureStrategy} that allows resuming the sequence.
+	 */
+	static OnNextFailureStrategy resumeIf(
+			Predicate<Throwable> causePredicate,
+			Consumer<Throwable> errorConsumer,
+			Consumer<Object>  valueConsumer) {
+		return new ResumeStrategy(causePredicate, errorConsumer, valueConsumer);
+	}
+
+	//==== IMPLEMENTATIONS ====
+	OnNextFailureStrategy STOP = new OnNextFailureStrategy() {
+
+		@Override
+		public boolean test(Throwable error, @Nullable Object value) {
+			return false;
+		}
+
+		@Override
+		public Throwable process(Throwable error, @Nullable Object value, Context context) {
+			Exceptions.throwIfFatal(error);
+			Throwable iee = new IllegalStateException("STOP strategy cannot process errors");
+			iee.addSuppressed(error);
+			return iee;
+		}
+	};
+
+	OnNextFailureStrategy RESUME_DROP = new ResumeDropStrategy(null);
+
+	final class ResumeStrategy implements OnNextFailureStrategy {
+
+		final Predicate<Throwable>  errorPredicate;
+		final Consumer<Throwable>   errorConsumer;
+		final Consumer<Object>      valueConsumer;
+
+		ResumeStrategy(@Nullable Predicate<Throwable> errorPredicate,
+				Consumer<Throwable> errorConsumer,
+				Consumer<Object> valueConsumer) {
+			this.errorPredicate = errorPredicate;
+			this.errorConsumer = errorConsumer;
+			this.valueConsumer = valueConsumer;
+		}
+
+		@Override
+		public boolean test(Throwable error, @Nullable Object value) {
+			return errorPredicate == null || errorPredicate.test(error);
+		}
+
+		@Override
+		@Nullable
+		public Throwable process(Throwable error, @Nullable Object value, Context context) {
+			if (errorPredicate == null) {
+				Exceptions.throwIfFatal(error);
```

Comment:
Nice that it's possible to match specific fatal errors with predicate and skip the default `throwIfFatal` :-)